server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1780,11 +1795,16 @@ public void deleteSegments(final Set<DataSegment> segments)
           public Void inTransaction(Handle handle, TransactionStatus transactionStatus)
           {
             int segmentSize = segments.size();
-            String dataSource = "";
+            String dataSource = segments.stream().findAny().map(DataSegment::getDataSource).orElse("?");
+
+            final String deleteFrom = StringUtils.format("DELETE from %s WHERE id = :id", dbTables.getSegmentsTable());
```

Comment:
```suggestion
            final String deleteSql = StringUtils.format("DELETE from %s WHERE id = :id", dbTables.getSegmentsTable());
```

Reply:
The primary key is sufficient, and assuming JDBI is doing the correct thing, using a single prepared statement with batching is going to be the most efficient way to get this done.